### PR TITLE
E2E for disabled localStorage

### LINKED ISSUE #2062: Add end-to-end tests with localStorage disabled
**Is your feature request related to a problem? Please describe.**
There have been two instances where localStorage being disabled or missing breaks the application. It is important that the app works with localStorage is disabled for sites who wish to embed Worldview into their own site (iFrame).

**Describe the solution you'd like**
Add an e2e test-suite with localStorage disabled.

**Describe alternatives you've considered**
None, but open to other ideas on how we could test this.

### PR DESCRIPTION
**Pull request by jasontk19**
## Description

Fixes #2062.

* create module `safeLocalStorage` to abstract `localStorage` calls preventing the need to check for availability before making calls to get/set items
* add eslint plugin to prevent use of `localStorage` (all calls should go through `safeLocalStorage` module instead)
* create Nightwatch envs for Chrome/Firefox with `localStorage` disabled
* tag a number of E2E tests (animation, tour) with `localStorageDisabled` that will be run both when `localStorage` is disabled/enabled to ensure some basic app functions work
* update `travis.yml` to run E2E tests for localStorage enabled and disabled for both Chrome and FF builds

@nasa-gibs/worldview

### CODE REVIEW

File: web/js/components/feature-alert/alert.js [hunk 1]
```diff
@@ -14,9 +14,7 @@ class FeaturedAlert extends React.Component {
   }
 
   dismissAlert() {
-    if (util.browser.localStorage) {
-      localStorage.setItem('<alert-name>', true);
-    }
+    // safeLocalStorage.setItem('<alert-name-goes-here>', true);
```
Comment (Benjaki2) [inline web/js/components/feature-alert/alert.js:17]: Do we want to `setItem` here? Also, I'm not sure that this file is still in use (Maybe it is still be useful for temporary alerts?).

File: web/js/containers/tour.js [hunk 2]
```diff
@@ -17,12 +19,18 @@ import { BULK_PALETTE_RENDERING_SUCCESS } from '../modules/palettes/constants';
 import { stop as stopAnimation } from '../modules/animation/actions';
 import { onClose as closeModal } from '../modules/modal/actions';
 import { layersParse12 } from '../modules/layers/util';
-import { endTour, selectStory, startTour } from '../modules/tour/actions';
+import {
+  endTour as endTourAction,
+  selectStory as selectStoryAction,
+  startTour as startTourAction,
+} from '../modules/tour/actions';
```
Comment (Benjaki2) [inline web/js/containers/tour.js:26]: 👍

File: web/js/modules/sidebar/reducers.js [hunk 3]
```diff
@@ -8,11 +8,10 @@ import {
...
   MOBILE_EXPAND_SIDEBAR,
 } from './constants';
-import util from '../../util/util';
+import safeLocalStorage from '../../util/local-storage';
 
-const wasCallapseRequested = util.browser.localStorage
-  ? localStorage.getItem('sidebarState') === 'collapsed'
-  : false;
+const { SIDEBAR_COLLAPSED } = safeLocalStorage.keys;
+const wasCallapseRequested = safeLocalStorage.getItem(SIDEBAR_COLLAPSED) === 'collapsed';
```
Comment (Benjaki2) [inline web/js/modules/sidebar/reducers.js:14]: preexisting misspell of `Collapse` in `wasCallapseRequested`.

File: web/js/components/feature-alert/alert.js (hunk 1 above)
Comment (jasontk19) [reply web/js/components/feature-alert/alert.js:17]: I kept it around for re-use in the event that we want to display another temporary "new feature" alert.  Currently it's not in use though.

**Pull request merged**